humility-dump-agent/src/udp.rs

```diff
@@ -79,6 +79,37 @@ impl<'a> UdpDumpAgent<'a> {
 
         Ok(reply)
     }
+
+    pub fn check_imageid(&mut self, image_id: &Vec<u8>) -> Result<()> {
+        let r = self.dump_remote_action(humpty::udp::Request::GetImageId)?;
+
+        match r {
+            Err(humpty::udp::Error::VersionMismatch { ours, theirs })
+                if ours == 2 && theirs > 2 =>
```

Comment:
Maybe add a comment here explaining that `ours` and `theirs` are swapped from our perspective (and/or an `assert_eq!(theirs, humpty::udp::version::CURRENT)`, to drive the point home)

Reply:
Heh -- look at the latest. ;)

Reply:
Added the additional check, but made it a `bail!` rather than a assert (because it points to a broken Hubris, not a broken Humility).